### DISCUSSION TITLE
Strange colorspace behaviour invloving heic reading and encoding.

### DISCUSSION THREAD

**Discussion by Risiamu** [libvips/libvips#2021] [Q&A]
Originally, i use ffmpeg to encode jpeg/png files to hevc frames and then box them with mp4box from GPAC project.
I have a collection of high resolution photos in heic format, which would be very power consuming on mobile device, so i thought i would use `vips resize` to reduce the resolution.

However, the resulted images are somehow darken than the original heic image from ffmpeg.

I ran a few tests with this original png.
![o](https://user-images.githubusercontent.com/30680793/88445543-f79c9e80-ce55-11ea-908b-c19ffc5e1284.png)

And this is the ffmpeg version heic "fffo.heic"(screen shot):
![图片](https://user-images.githubusercontent.com/30680793/88445553-0e42f580-ce56-11ea-8711-58055daf5800.png)

When i use `vips heifsave fffo.heic fffo2.heic` the resulted fffo2.heic is darkened.
![图片](https://user-images.githubusercontent.com/30680793/88445595-3df1fd80-ce56-11ea-8edd-fb2cee9d1be7.png)

However, if i run the above operation with heic converted from vips (filename vfo.heic), everything works fine.
![图片](https://user-images.githubusercontent.com/30680793/88445631-64b03400-ce56-11ea-9c19-c4241b6b510f.png)

I use `magick identify` on heics converted from png (with ffmpeg and vips) and the output show no difference.
```
PS D:\IMAGES\SX\2020-07> magick identify fffo.heic
fffo.heic HEIC 358x476 358x476+0+0 8-bit YCbCr 0.000u 0:00.000
PS D:\IMAGES\SX\2020-07> magick identify vfo.heic
vfo.heic HEIC 358x476 358x476+0+0 8-bit YCbCr 0.000u 0:00.000
PS D:\IMAGES\SX\2020-07>
```
I upload my images:
[2020-07.zip](https://github.com/libvips/libvips/files/4974726/2020-07.zip)
fffo.heic `ffmpeg converted from original png`
fffo2.heic `vips heifsave on the ffmpeg version heic`
o.png `origin png file`
vfo.heic `vips heifsave on the original png`
vfo2.heic `vips heifsave on the vfo.heic`

BTW, the command i use to convert files to heic.
```
		ffmpeg -i "$($FilePath)"  -crf $CRTValue -vf "crop=trunc(iw/2)*2:trunc(ih/2)*2" -pix_fmt yuv420p -sws_flags spline+accurate_rnd+full_chroma_int -color_range 1 -colorspace 5 -color_primaries 5 -color_trc 6   -f hevc "$($FileNoExtensionPath).hvc";
		MP4Box -add-image "$($FileNoExtensionPath).hvc:primary" -ab heic -new "$($FileNoExtensionPath).heic";
```

**Comment by kleisauke** [comment]
I could reproduce this with libheif master (https://github.com/strukturag/libheif/commit/b29666747b4be5fcaf3f17a551ed5c2fd596872b) and libde265 v1.0.5 when doing:
```bash
$ heif-convert fffo.heic x.png
```

However, I could not reproduce this with the online decoder from:
https://strukturag.github.io/libheif/

In an attempt to bisect this, I tried libheif v1.3.2 and libde265 v1.0.2 with this Dockerfile:
<details>
   <summary>Details</summary>

```Dockerfile
FROM centos:7

# Import archive signing keys and update packages
RUN rpm --import https://dl.fedoraproject.org/pub/epel/RPM-GPG-KEY-EPEL-7 && \
    yum update -y

# Install libheif v1.3.2
RUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
    yum install -y https://download1.rpmfusion.org/free/el/rpmfusion-free-release-7.noarch.rpm && \
    rpmkeys --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rpmfusion-free-el-7 && \
    yum install -y --setopt=tsflags=nodocs \
        libheif-1.3.2 && \
    yum clean all

VOLUME /data
ENTRYPOINT ["/usr/bin/heif-convert"]
```

Usage:
```bash
docker build -t heif-convert .
docker run -it -v $(pwd):/data heif-convert /data/fffo.heic /data/x.png
```
</details>

But the gray background is still present. Assuming it's in libheif, a git bisect from https://github.com/strukturag/libheif/commit/fb60ff7f50a99bee70e28e9b2d4c1e74f76c3ea6 to https://github.com/strukturag/libheif/tree/v1.3.2 should reveal which commit introduces this behavior (I'm not sure if it's a bug).

**Comment by Risiamu** [comment]
@kleisauke  I think will open another issue there.
